### PR TITLE
ci: deactivate dependabot previews

### LINKED ISSUE #2505: Dependabot preview builds are failing
### Steps to reproduce the problem

1. Go to the [project actions](https://github.com/Qiskit/qiskit.org/actions)
2. Check the "Preview" workflows by "dependabot"
3. See how they fail to generate the preview

### What is the current behavior?

The preview builds form "dependabot" PRs are failing.

### What is the expected behavior?

The preview builds should be generated.

### Screenshots (optional but very helpful)

Screenshot 2022-02-22 at 10 43 14

### PR DESCRIPTION
**Pull request by eddybrando**
## Changes

Avoid running the workflows to generate preview deployments for "dependabot" PRs.

Closes #2505
Co-authored-by: @abdonrd 

## Implementation details

Added a check to the workflows to avoid running if the author is "dependabot".

### CODE REVIEW

**Review by Tansito — DISMISSED** [review]
I'm agree with no running the preview action for PRs dependencies and I'm open to study it for platypus too.

File: .github/workflows/preview-cleanup.yml [hunk 1]
```diff
@@ -6,9 +6,8 @@ on:
 
 jobs:
   teardown:
-    # Run only from the original repository
-    # Because this job requires secrets, which cannot be shared with forks
-    if: github.event.pull_request.head.repo.full_name == github.repository
+    # This job requires secrets, which cannot be accessed by forks or dependabot
+    if: ${{ github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot' }}
```
Comment (abdonrd) [inline .github/workflows/preview-cleanup.yml:10]:
```suggestion
    if: ${{ github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]' }}
```

File: .github/workflows/preview-deploy.yml [hunk 2]
```diff
@@ -5,9 +5,8 @@ on:
 
 jobs:
   setup:
-    # Run only from the original repository
-    # Because this job requires secrets, which cannot be shared with forks
-    if: github.event.pull_request.head.repo.full_name == github.repository
+    # This job requires secrets, which cannot be accessed by forks or dependabot
+    if: ${{ github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot' }}
```
Comment (abdonrd) [inline .github/workflows/preview-deploy.yml:9]:
```suggestion
    if: ${{ github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]' }}
```

**Review by abdonrd — CHANGES_REQUESTED** [review]
If I remember correctly, the actor should be `dependabot[bot]`

**Review by abdonrd — APPROVED** [review]

**Pull request merged**